### DISCUSSION TITLE
How to change sort order of the gallery?

### DISCUSSION THREAD

**Discussion by learningchannel** [datenstrom/community#141] [Ask a question] [answered]
hi i'd like to know if there is a way to easily change sort order (ex: ascending or descending) within the [gallery] shortcut. please explain what would be the easiest way to be able to do this? great cms.  thanks in advance

**Comment by markseuffert** [comment]
Sorry, the [gallery plugin](https://github.com/annaesvensson/yellow-gallery) shows pictures as they are shown in the file manager, in natural sort order. The easiest way to change the sort order is to rename your image files, for example with a numeric prefix.

**Comment by learningchannel** [comment]
thank you!

**Comment by markseuffert** [comment]
Just ask if there are more questions. :smile:

**Comment by mymurkas** [comment]
I also would like options to sort the gallery in other ways. When having a lot of images it takes some time to re-name them all. Could this be done?

**Reply by markseuffert** [reply]
You can put them in different folders, see also datenstrom/yellow#147.

**Reply by mymurkas** [reply]
Thank you for a quick reply! 
This is what I have done. Separate folders. But one thing is when to add new images and to have them first in the gallery. One have to keep adding zeros to the filnames?

**Reply by markseuffert** [reply]
One shouldn't have to suffer that much. Do you have a little more context, for example problem you want to solve, what kind of website, for yourself or an organisation, your programming background?

**Reply by mymurkas** [reply]
I'm makeing a site for an artist who wants a archive of images. One big gallery of images. and also to be able to add new images, and they should show first, on the top. So now I did put all images in one folder anyway.
I want to keep it simple for him to add files and to be able to just add a number on the file name would be the easiest I guess.

**Reply by mymurkas** [reply]
I don't have much programing background. Mostly copy/paste php :)

**Reply by markseuffert** [reply]
No, this is not possible with this (little) extension. 

This would require making a new extension with the [API for developers](https://datenstrom.se/yellow/help/api-for-developers). As a developer you can make something that shows new images sorted by date, something like Apple's Photo Stream or Unsplash's image explorer. Maybe with an option to search for image captions? Maybe with an option to put the best shots on top? Maybe with the month over a collection of images? The API has you covered. 😁

**Reply by markseuffert** [reply]
As a starting point for new developers, you can make a custom gallery extension and then tell it to sort files by date.

Replace `$files = $this->yellow->media->index()->match("#$images$pattern#");` 
with `$files = $this->yellow->media->index()->match("#$images$pattern#")->sort("modified", false);` 

For more information see the API. Hope this helps.

**Reply by mymurkas** [reply]
Thanks for the response! I'll look into it. Would be nice with an extention, perhaps not as far as Apple's Photo Stream or Unsplash's image explorer but something :)

**Comment by schulle4u** [comment, marked as answer]
The [gallery extension](https://github.com/annaesvensson/yellow-gallery) now supports different sorting methods (e.g. name, modified, size). The update procedure should take care of updating your shortcuts, however it is always wise to check manually.
